src/workers/notifications/commentDeleted.ts

```diff
@@ -0,0 +1,47 @@
+import { messageToJson, Worker } from '../worker';
+import { Notification, Comment } from '../../entity';
+import { ChangeObject } from '../../types';
+
+interface Data {
+  comment: ChangeObject<Comment>;
+}
+
+const worker: Worker = {
+  subscription: 'api.comment-deleted-notification-cleanup',
+  handler: async (message, con, logger) => {
+    const data: Data = messageToJson(message);
+    const { comment } = data;
+
+    const databaseComment = await con
+      .getRepository(Comment)
+      .findOneBy({ id: comment?.id });
+    if (!databaseComment) {
+      return logger.error(
+        {
+          data,
+          messageId: message.messageId,
+        },
+        'failed to delete notifications due to comment deletion',
+      );
+    }
```

Comment:
This is not what I meant because at this point the comment is already not in the database as we're dealing with deleted comments. We only need an if statement to check if comment?.id is defined